org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/RefactorProcessor.java

```diff
@@ -28,18 +28,23 @@
 
 	private PreferenceManager preferenceManager;
 
+	private InvertBooleanSubProcessor invertBooleanSubProcessor;
+
 	public RefactorProcessor(PreferenceManager preferenceManager) {
 		this.preferenceManager = preferenceManager;
+		invertBooleanSubProcessor = new InvertBooleanSubProcessor(this.preferenceManager);
```

Comment:
Given InvertBooleanSubProcessor is stateless, could you make it to a utility class, e.g. InvertBooleanUtility.java?

And the only instance method `getInverseLocalVariableProposals` which used preferenceManager is not necessarily in this class.

Reply:
Yes, you are right. I can make the class stateless, and move the instance method elsewhere. But I might keep the name as `InvertBooleanSubProcessor`. There are a lot of `XXXSubProcessor`, all stateless. So I'd like to keep the consistency.

Reply:
@testforstephen Alright, I've renamed it to `InvertBooleanUtility` to avoid ambiguity.